core/parse-xml.c

```diff
@@ -436,10 +436,10 @@ static void event_name(char *buffer, char *name)
 const char *libdc_divemode_text[] = { "oc", "cc", "pscr", "freedive", "gauge"};
```

Comment:
I see how you'd switch from cc to oc on a bailout. Some of the other possible transitions seem odd to me. But maybe that just means they aren't possible at all (e.g. oc->freedive)

Reply:
There is also PSCR (which might require a bailout) and maybe it was for training or you fixed a problem so it also makes sense to go back on the loop. I agree that other combinations (e.g. PSCR<->CCR) will have little practical importance.